src/libstd/lib.rs

```diff
@@ -365,6 +368,8 @@ pub mod prelude;
 // Public module declarations and re-exports
 #[stable(feature = "rust1", since = "1.0.0")]
 pub use core::any;
+#[stable(feature = "simd_arch", since = "1.27.0")]
+pub use core::arch;
```

Comment:
This is the largest change here. Instead of re-compiling core::arch again and including it two times, one in std and one in core, this PR changes libstd to re-export the intrinsics from core directly.